Delete NetEventSource.Enter/Exit events

stephentoub opened the pull request:
They're super verbose, very inconsistent, costly, and not helpful (they appear to be trying to provide tracing of the full flow of calls, but they fail to do even that).

@geoffkizer, any concerns?

Contributes to https://github.com/dotnet/runtime/issues/38754
cc: @alnikola, @MihaZupan, @dotnet/ncl

wfurt commented on src/libraries/Common/src/Interop/Windows/SspiCli/SSPIWrapper.cs:
```diff
@@ -356,24 +344,18 @@ private static unsafe int EncryptDecryptHelper(OP op, ISSPIInterface secModule,
             int errorCode = secModule.QueryContextChannelBinding(securityContext, contextAttribute, out result);
             if (errorCode != 0)
             {
-                if (NetEventSource.IsEnabled) NetEventSource.Exit(null, $"ERROR = {ErrorDescription(errorCode)}");
```
Should we turn this to NetEventSource.Error?

wfurt commented on src/libraries/Common/src/Interop/Windows/SspiCli/SSPIWrapper.cs:
```diff
@@ -391,21 +373,12 @@ private static unsafe int EncryptDecryptHelper(OP op, ISSPIInterface secModule,
             {
-                if (errorCode != 0)
-                {
-                    if (NetEventSource.IsEnabled) NetEventSource.Exit(null, $"ERROR = {ErrorDescription(errorCode)}");
```
NetEventSource.Error ?

wfurt commented on src/libraries/Common/src/Interop/Windows/SspiCli/SSPIWrapper.cs:
```diff
@@ -453,20 +418,15 @@ private static unsafe int EncryptDecryptHelper(OP op, ISSPIInterface secModule,
                 }
 
-                string? result = Marshal.PtrToStringUni(sspiHandle.DangerousGetHandle());
-                if (NetEventSource.IsEnabled) NetEventSource.Exit(null, result);
```
NetEventSource.Info ?

wfurt commented on src/libraries/Common/src/System/Net/NTAuthentication.Common.cs:
```diff
@@ -101,8 +101,6 @@ internal NTAuthentication(bool isServer, string package, NetworkCredential crede
         [MemberNotNull(nameof(_package))]
         private void Initialize(bool isServer, string package, NetworkCredential credential, string? spn, ContextFlagsPal requestedContextFlags, ChannelBinding? channelBinding)
         {
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, package, spn, requestedContextFlags);
```
I think we should log at least spn as info. (unless we do that somewhere else).

wfurt commented on src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs:
```diff
@@ -39,15 +38,11 @@ public static IPHostEntry GetHostEntry(IPAddress address)
 
-            IPHostEntry ipHostEntry = GetHostEntryCore(address);
-
-            if (NetEventSource.IsEnabled) NetEventSource.Exit(address, $"{ipHostEntry} with {ipHostEntry.AddressList.Length} entries");
```
I would leave it in as Info so we can know how names were resolved.

stephentoub replied:
I'm putting it back as Info per your request, but this is a great example of how inconsistent the current logging is: while this overload logs the result, many of the others (including most of the async ones) don't.

wfurt replied:
yes, there is lot of inconsistencies. I'm wondering if we should proceed and open separate issue for each area to add what we think is useful. It may be more work but we may end up with better output.

stephentoub replied:
> I'm wondering if we should proceed and open separate issue for each area to add what we think is useful.

That's fine.  Honestly I'd be fine deleting all of inherited tracing and starting over.

wfurt commented on src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs:
```diff
@@ -918,26 +875,16 @@ internal SecurityStatusPal Encrypt(ReadOnlyMemory<byte> buffer, ref byte[] outpu
-            {
-                if (NetEventSource.IsEnabled)
-                    NetEventSource.Exit(this, $"ERROR {secStatus}");
-            }
```
I would leave it as NetEventSource.Error

wfurt commented on src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs:
```diff
@@ -981,8 +925,6 @@ internal bool VerifyRemoteCertificate(RemoteCertValidationCallback? remoteCertVa
 
                 if (remoteCertificateEx == null)
                 {
-                    if (NetEventSource.IsEnabled)
```
NetEventSource.Error?

wfurt commented on src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:
```diff
@@ -85,8 +85,6 @@ public Socket(SocketType socketType, ProtocolType protocolType)
         // Initializes a new instance of the Sockets.Socket class.
         public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType)
         {
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, addressFamily);
```
should this be info we can can track socket creation?

wfurt commented on src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:
```diff
@@ -792,8 +789,6 @@ internal bool CanTryAddressFamily(AddressFamily family)
         // Associates a socket with an end point.
         public void Bind(EndPoint localEP)
         {
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, localEP);
```
Info?

wfurt commented on src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:
```diff
@@ -1071,7 +1043,6 @@ public void Close(int timeout)
         /// <param name="backlog">The maximum length of the pending connections queue.</param>
         public void Listen(int backlog)
         {
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, backlog);
```
Info?

wfurt commented on src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs:
```diff
@@ -31,8 +31,6 @@ public TcpClient() : this(AddressFamily.Unknown)
         // Initializes a new instance of the System.Net.Sockets.TcpClient class.
         public TcpClient(AddressFamily family)
         {
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, family);
```
Info?

wfurt commented on src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs:
```diff
@@ -60,16 +55,12 @@ public TcpClient(IPEndPoint localEP)
         // the specified host.
         public TcpClient(string hostname, int port)
         {
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, hostname);
```
Info?

wfurt commented on src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPListener.cs:
```diff
@@ -20,21 +20,18 @@ public class TcpListener
         // Initializes a new instance of the TcpListener class with the specified local end point.
         public TcpListener(IPEndPoint localEP)
         {
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, localEP);
```
Info?

wfurt reviewed (commented):
While I did not see Enter/Exit particularly useful, some of the existing messages do carry useful information - error details, object creation or significant operations. It may be worth of converting some into relevant entry. It would be nice IMHO if tracing gives us general visibility to what some app does. This is important for cases like Azure where we cannot get debugger or even packet captures.

stephentoub committed:
Delete NetEventSource.Enter/Exit events

They're super verbose, very inconsistent, costly, and not helpful (they appear to be trying to provide tracing of the full flow of calls, but they fail to do even that).
